app/post.go

```diff
@@ -1205,7 +1205,11 @@ func (a *App) GetPostsForChannelAroundLastUnread(c request.CTX, channelID, userI
 	}
 	// Reset order to only include the last unread post: if the thread appears in the centre
 	// channel organically, those replies will be added below.
-	postList.Order = []string{lastUnreadPostId}
+	postList.Order = []string{}
+	// Add lastUnreadPostId in order, only if it hasn't been filtered as per the cloud plan's limit
+	if _, ok := postList.Posts[lastUnreadPostId]; ok {
+		postList.Order = []string{lastUnreadPostId}
+	}
```

Comment:
Not directly related to this PR, but if the lastUnreadPostId has been filtered by the cloud plan's limit, can't we just skip the `a.GetPostsBeforePost` call below? 
If the `lastUnreadPostId` is not accessible, then `postListBefore` won't be accessible, and so we can just skip the app and DB call.

Reply:
Yes, you're right, we can definitely avoid this unnecessary call.